src/App.js

```diff
@@ -14,11 +14,68 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    good: 'Jam',
+    selected: true,
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <h1>
+          Selected good: -
+          {' '}
+          {this.state.selected
+            ? `${this.state.good} is selected`
+            : 'No goods selected'
+          }
+          {' '}
+          {this.state.selected
+            ? (
+              <button
+                type="button"
+                onClick={() => {
+                  this.setState({
+                    good: null,
+                    selected: false,
+                  });
+                }}
+              >
+                X
+              </button>
+            )
+            : ''
```

Comment:
if you don't need to show anything in one of the cases don't use the ternary operator, use `&&`